### ISSUE TITLE
Docker base image not available in linux/arm/v7 variant

### ISSUE THREAD

**Issue by CodeWithMa** [krateng/maloja#229]
The Docker image used as the base for Maloja is currently not available in the `linux/arm/v7` variant. This has been confirmed by checking the [lsiobase/alpine tags](https://hub.docker.com/r/lsiobase/alpine/tags?page=1&name=3.17) where the specific variant we rely on is missing.

Consequently, the current build configuration, as seen [here](https://github.com/krateng/maloja/blob/39a42e915c3f598ece2a4a70c7ccd256be7531ba/.github/workflows/docker.yml#L57), will likely break in the future if the issue isn't addressed.

Drop support for linux/arm/v7?

**Comment by krateng** [comment]
Oof, that's kind of annoying as RasPis usually are in use for a long time. But I guess there is no choice. Any specific reason they don't offer a v7 image?

**Comment by FoxxMD** [comment]
Just because lsio does not build armv7 images does not mean there is any reason we can't. Alpine certainly supports [armv7](https://hub.docker.com/r/arm32v7/alpine/).

In my experience alpine armv7 variant issues are usually an issue of a project dependency that relies on a c-compiled binary not having a pre-built version for musl. This usually only occurs for binaries that aren't standard os libraries or already in the [main/community](https://pkgs.alpinelinux.org/packages) alpine package repos.

Even in these cases they can be compiled from source since alpine provides all the standard dev libs but sometimes there can be differences between musl/gcc that cause source builds to fail.

Since the current image builds fine and maloja is mature software with dependencies that aren't likely to change dramatically the chances of running into build issues in the future is pretty low. Additionally, the docker image is pretty straightforward. It can always be ported to a `debian-slim` OS base if this issue does arise.

**Comment by krateng** [comment]
Alright I've removed the arch from the latest image. Is there any cool way to just tell the build command to build all missing base images as well? Or do I need convoluted extra instructions for that?

**Comment by duckfromdiscord** [comment]
> Drop support for linux/arm/v7?

Probably a bad idea. I'd say Raspberry Pi is widely used enough among the userbase of maloja (that is, those who want to be able to host their data locally) that we should keep support for it.

Until we can deal with the root issue, why not stick with an older version of lsiobase/alpine that supports armv7? Or swap to a different base entirely for armv7? @FoxxMD is right that, worst case, we can just change the distro to Debian or something else like it.

**Comment by krateng** [comment]
The latest version that is built for arm/v7 seems to be 3.13, which has reached EOL a year ago. lsiobase seems to have dropped arm/v7 for its other images like debian as well?

**Comment by CodeWithMa** [comment]
> The latest version that is built for arm/v7 seems to be 3.13, which has reached EOL a year ago. lsiobase seems to have dropped arm/v7 for its other images like debian as well?

Yes they dropped support for all their images: https://www.linuxserver.io/blog/a-farewell-to-arm-hf

**Comment by duckfromdiscord** [comment]
That's upsetting. There must be plenty of users who still have their Pi and want to use it. I guess we have to find a different container from someone else entirely.

**Comment by FoxxMD** [comment]
The majority of [Raspi hardware is still supported.](https://wiki.debian.org/RaspberryPi#Raspberry_Pi_Models) 

The pi 3/a/b, Zero 2, 4, and 400 all run arm64. 

Pi 1 and 2 (released 2014 and older), Pico, and the Zero are armhf. All of these systems have less than 1gb memory as well. The pico is 264kb, zero has 512mb. You'd be hard pressed to run a modern OS with docker on any of them.

**Comment by duckfromdiscord** [comment]
> I'd be willing to build an armhf containerfile if someone who is already using maloja on armhf hardware in docker can chime in that they are no longer able to upgrade due to the arch being dropped.

That sounds fair to me. It is true that the Pi 1 and 2 are pretty old and have very little memory. I own what I believe is a Pi Zero but I can't say I use maloja on it.
